lib/model/Nutriments.dart

```diff
@@ -5,109 +5,131 @@ import '../interface/JsonObject.dart';
 
 part 'Nutriments.g.dart';
 
+// TODO(monsieurtanuki); make some fields read-only, like Units and nova.
 /// Values in nutrients
 @JsonSerializable()
 class Nutriments extends JsonObject {
-  /// Salt, in grams, per 100 grams of food
+  /// Salt, in grams, per 100 grams of product as sold
   @JsonKey(
       name: 'salt_100g', includeIfNull: false, fromJson: JsonObject.parseDouble)
   double? salt;
 
-  /// Fibers, in grams, per 100 grams of food
+  /// Fibers, in grams, per 100 grams of product as sold
   @JsonKey(
       name: 'fiber_100g',
       includeIfNull: false,
       fromJson: JsonObject.parseDouble)
   double? fiber;
 
-  /// Sugars, in grams, per 100 grams of food
+  /// Sugars, in grams, per 100 grams of product as sold
   @JsonKey(
       name: 'sugars_100g',
       includeIfNull: false,
       fromJson: JsonObject.parseDouble)
   double? sugars;
 
-  /// Fats, in grams, per 100 grams of food
+  /// Fats, in grams, per 100 grams of product as sold
   @JsonKey(
       name: 'fat_100g', includeIfNull: false, fromJson: JsonObject.parseDouble)
   double? fat;
 
-  /// Saturated Fats, in grams, per 100 grams of food
+  /// Saturated Fats, in grams, per 100 grams of product as sold
   @JsonKey(
       name: 'saturated-fat_100g',
       includeIfNull: false,
       fromJson: JsonObject.parseDouble)
   double? saturatedFat;
 
-  /// Proteins, in grams, per 100 grams of food
+  /// Proteins, in grams, per 100 grams of product as sold
   @JsonKey(
       name: 'proteins_100g',
       includeIfNull: false,
       fromJson: JsonObject.parseDouble)
   double? proteins;
 
-  /// Nova Group
+  /// Nova Group per 100g (?)
   @JsonKey(
       name: 'nova-group_100g',
       includeIfNull: false,
       fromJson: JsonObject.parseInt)
   int? novaGroup;
 
-  /// Energy, in kJ, per 100 grams of food
+  /// Energy (?), in kJ, per 100 grams of product as sold
```

Comment:
This field is in kJ. It is equal to energy-kj_100g if we have it, or computed from energy-kcal_100g otherwise.

Reply:
I'm actually writing a comment on #567: a source of confusion is that we have fields for both `'energy'` and `'energy-kj'`.
Again, I can understand that for historical reasons both fields are there on the server side, but from a user perspective it's quite annoying to know which field we should use. And I would like to make it more user-friendly on the dart side.
I suggest that we get rid of one of them (in dart). I would prefer to keep `'energy-kj'`, as I've understood that's an input value (not a computed value), as long as it's OK when we read from/write to the server. And maybe make the `energy` and `energyServing` dynamic fields, like
```dart
double get energy => energyKj100g ?? energyKCal100g * 4.184;
```